### ISSUE TITLE
Crash on 19.1

### ISSUE THREAD

**Issue by hamiltont** [nzbget/nzbget#531]
After replacing 17.0 with 19.1, I now receive a SIGABRT soon after launch (ubuntu 16.04). 

After replacing my custom config file with the default 19.1 config, 19.1 loaded fine without crashing. I then restored by 17.0 config using the GUI restore method, and was able to see all settings. However, as soon as I clicked "save", I again saw the SIGABRT. 

I then compiled 19.1 with debug symbols, and captured the backtrace shown below. 

If this is not enough info to identify the issue I'm happy to gather more, but it may take me a few days to respond as I'm currently traveling 

```

$ gdb --args nzbget --configfile=/etc/nzbget.conf --server --option OutputMode=loggable
<snip>
(gdb) run
Starting program: /usr/bin/nzbget --configfile=/etc/nzbget.conf --server --option OutputMode=loggable
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff3c10700 (LWP 29391)]
[New Thread 0x7ffff340f700 (LWP 29392)]
[New Thread 0x7ffff2c0e700 (LWP 29393)]
[New Thread 0x7ffff240d700 (LWP 29394)]
[New Thread 0x7ffff1c0c700 (LWP 29395)]
[New Thread 0x7ffff140b700 (LWP 29396)]
[New Thread 0x7ffff0c0a700 (LWP 29397)]
[New Thread 0x7fffebfff700 (LWP 29398)]
[New Thread 0x7fffeb7fe700 (LWP 29406)]
[New Thread 0x7fffeaffd700 (LWP 29407)]
[New Thread 0x7fffea3da700 (LWP 29408)]
[New Thread 0x7fffe9bd9700 (LWP 29409)]
terminate called after throwing an instance of 'std::out_of_range'
  what():  vector::_M_range_check: __n (which is 17) >= this->size() (which is 17)

Thread 4 "nzbget" received signal SIGABRT, Aborted.
[Switching to Thread 0x7ffff2c0e700 (LWP 29393)]
0x00007ffff61c0428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
54  ../sysdeps/unix/sysv/linux/raise.c: No such file or directory.

(gdb) bt full
#0  0x00007ffff61c0428 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
        resultvar = 0
        pid = 28850
        selftid = 28857
#1  0x00007ffff61c202a in __GI_abort () at abort.c:89
        save_stage = 2
        act = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, sa_mask = {__val = {0, 0, 140737322695679, 140737326155072, 1, 10, 140737323213533, 10, 140737326159728, 140737266116320,
              140737323213533, 1, 10004912, 140737027079536, 140737326159728, 140737266116352}}, sa_flags = -159418240, sa_restorer = 0x7fffe4817e18}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x00007ffff67fa84d in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#3  0x00007ffff67f86b6 in ?? () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#4  0x00007ffff67f8701 in std::terminate() () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#5  0x00007ffff67f8919 in __cxa_throw () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#6  0x00007ffff68213f7 in std::__throw_out_of_range_fmt(char const*, ...) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
No symbol table info available.
#7  0x00000000004ba4cb in std::vector<std::unique_ptr<ArticleInfo, std::default_delete<ArticleInfo> >, std::allocator<std::unique_ptr<ArticleInfo, std::default_delete<ArticleInfo> > > >::_M_range_check
    (this=<optimized out>, this=<optimized out>, __n=17) at /usr/include/c++/5/bits/stl_vector.h:803
No locals.
#8  std::vector<std::unique_ptr<ArticleInfo, std::default_delete<ArticleInfo> >, std::allocator<std::unique_ptr<ArticleInfo, std::default_delete<ArticleInfo> > > >::at (__n=17, this=<optimized out>)
    at /usr/include/c++/5/bits/stl_vector.h:824
No locals.
#9  DiskState::LoadFileState (this=this@entry=0x7e2790, fileInfo=fileInfo@entry=0x7fffe418a710, servers=servers@entry=0x7df0e8, infile=..., formatVersion=2, completed=completed@entry=true)
    at daemon/queue/DiskState.cpp:1132
        pa = <optimized out>
        statusInt = 2
        status = <optimized out>
        i = 17
        hasArticles = true
        successArticles = 62
        failedArticles = 7
        High1 = 0
        Low1 = 0
        High2 = 0
        Low2 = 48603731
        High3 = 0
        Low3 = 5553205
        buf = "\005", '\000' <repeats 15 times>, "\001\000\000\000\000\000\000\000h\341\377\367\377\177\000\000\300\344\377\367\377\177\000\000\260\314\300\362\377\177\000\000\b\000\000\000\001", '\000' <repeats 11 times>, "\300\344\377\367\377\177\000\000\316f \366\377\177\000\000P\322\300\362\377\177\000\000@\322\300\362\377\177\000\000\061\270R\000\000\000\000\000\300\323\300\362\377\177\000\000\000\000\000\000\000\000\000\000-\211\035\366\377\177\000\000\350\fT\000\000\000\000\000\340\323\300\362\377\177\000\000\000\000\000\000\000\000\000\000\270\315\300\362\377\177\000\000\250\315\300\362\377\177", '\000' <repeats 14 times>, "\377\177\000\000\300\315\300\362\377\177\000\000\000\000\000\000\377\177", '\000' <repeats 18 times>...
        completedArticles = 17
        size = 69
#10 0x00000000004ba579 in DiskState::LoadFileState (this=0x7e2790, fileInfo=0x7fffe418a710, servers=0x7df0e8, completed=completed@entry=true) at daemon/queue/DiskState.cpp:1084
        __FUNCTION__ = "LoadFileState"
        filename = {
          m_data = "3349c", '\000' <repeats 11 times>, "@>\225\344\377\177\000\000P\021T\000\000\000\000\000]\000\000\000\000\000\000\000\344CP", '\000' <repeats 13 times>, " \000\000\000\060\000\000\000\320\331\300\362\377\177\000\000\000\331\300\362\377\177\000\000Starting Thread (Thr"}
        stateFile = {m_destFilename = {
            m_data = "/var/opt/nzbget/queue/3349c\000\377\177\000\000Creating ArticleDownloader (ArticleDownloader.cpp:34:ArticleDownloader)\000rdinator.cpp:576:StartArticleDownload)\000\000\377\003\000\000\000\000\000\000\226\030*\366\377\177\000\000\360\321\300\362\377\177\000\000\360\321\300\362\377\177\000\000\340\321\300\362\377\177\000\000\340\325\300\362\377\177\000\000\340\321\300\362\377\177\000\000\000"...}, m_tempFilename = {m_data = "/var/opt/nzbget/queue/3349c", '\000' <repeats 996 times>}, m_transactional = false, m_formatVersion = 5, m_fileVersion = 2, m_file = {<DiskFile> = {
              m_file = 0x7fffe4816b20}, <No data fields>}}
        infile = <optimized out>
#11 0x00000000004d0334 in QueueCoordinator::LoadPartialState (this=<optimized out>, fileInfo=0x7fffe418a710) at daemon/queue/QueueCoordinator.cpp:898
        outputFilename = {m_data = '\000' <repeats 980 times>...}
#12 0x00000000004d207d in QueueCoordinator::GetNextArticle (this=this@entry=0x7e1810, downloadQueue=downloadQueue@entry=0x7e1838, fileInfo=@0x7ffff2c0dea8: 0x7fffe418a710,
    articleInfo=@0x7ffff2c0deb0: 0x7fffe4953e40) at daemon/queue/QueueCoordinator.cpp:518
        checkedFiles = std::vector of length 1, capacity 100 = {0x7fffe418a610}
        curDate = 1523404432
#13 0x00000000004d21ca in QueueCoordinator::Run (this=0x7e1810) at daemon/queue/QueueCoordinator.cpp:196
        downloadQueue = {m_ptr = 0x7e1838, m_mutex = 0x7e18f8}
        hasMoreArticles = <optimized out>
        fileInfo = 0x7fffe418a710
        articleInfo = 0x7fffe4953e40
        freeConnection = false
        downloadsChecked = false
        downloadStarted = false
        connection = 0x7e4b40
        standBy = <optimized out>
        sleepInterval = <optimized out>
        __FUNCTION__ = "Run"
        wasStandBy = false
        articeDownloadsRunning = <optimized out>
        resetCounter = 0
#14 0x000000000050a125 in Thread::thread_handler (object=0x7e1810) at daemon/util/Thread.cpp:168
        __FUNCTION__ = "thread_handler"
        thread = 0x7e1810
#15 0x00007ffff7bc16ba in start_thread (arg=0x7ffff2c0e700) at pthread_create.c:333
        __res = <optimized out>
        pd = 0x7ffff2c0e700
        now = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {140737266116352, -1309122655594351830, 0, 140737488348191, 140737266117056, 0, 1309116343091589930, 1309105590867063594}, mask_was_saved = 0}},
          priv = {pad = {0x0, 0x0, 0x0, 0x0}, data = {prev = 0x0, cleanup = 0x0, canceltype = 0}}}
        not_first_call = <optimized out>
        pagesize_m1 = <optimized out>
        sp = <optimized out>
        freesize = <optimized out>
        __PRETTY_FUNCTION__ = "start_thread"
#16 0x00007ffff629241d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
No locals.
```

**Comment by hugbug** [comment]
It crashes when loading queue disk state (from QueueDir). As a test remove the directory (make a backup) a try to start.

The error occurs on file "3349c". You can try to remove only it and test if the program starts. This file isn't very important :)

Also it would be great if you could test with v20.0-testing as it contains many fixes, although I can't remember a particular one related to queue loading.

If you are OK with sharing your queue and history data with me I could try to debug (and fix) the issue - in that case please send me the content of QueueDir to nzbget@gmail.com.

**Comment by hamiltont** [comment]
Confirmed that removing queuedir fixed the crash. Removing 3349c file causes a (different) error on 19.1. Switching to 20.0, I get similar errors (either from 3349c or, if I remove that one file, a different error). Seems this entire queuedir from 17 is hitting edge case problems, so I've emailed you the contents - about 30MB - in hopes it helps you debug. 

I'll close this ticket for now - just emptying the directory is enough to get me working with 19.1, I'm happy enough with that :-) 

Thanks for the help
